alacritty/src/event.rs

```diff
@@ -456,13 +455,22 @@ impl<N: Notify> Processor<N> {
                 },
                 Event::ConfigReload(path) => {
                     processor.ctx.message_buffer.remove_target(LOG_TARGET_CONFIG);
-                    processor.ctx.resize_pending.message_buffer = Some(());
+                    processor.ctx.display_update_pending.message_buffer = Some(());
 
                     if let Ok(config) = config::reload_from(&path) {
+                        if processor.ctx.font.size == processor.config.font.size {
+                            processor.ctx.display_update_pending.font = Some(FontUpdate::Reset);
+                        }
                         processor.ctx.terminal.update_config(&config);
 
-                        if *processor.ctx.font_size == processor.config.font.size {
-                            processor.ctx.resize_pending.font_size = Some(FontResize::Reset);
+                        // use copy of new config with font size overridden to match the current
+                        // font size in order to determine if any font value other than size has
+                        // changed
+                        let mut font_copy = config.font.clone();
+                        font_copy.size = processor.ctx.font.size;
```

Comment:
The necessity of the comment should already indicate that this isn't really a great solution to this problem, it might be easier to resolve this once the other issues are addressed, but I thought I'd point it out now already so you can start working if you have an idea.

Reply:
There are two alternatives I can think of:

* write a one-off function that compares every field (except for size) of the processor font with the config font to determine if anything other than the size has changed; the problem with this approach in my opinion is that it leaves us open to missing future additions to the `Font` struct
* write a procedural macro to make this function for us, parameterizing it to exclude the `size` field (i'm not certain this is possible, just vaguely suspect it might be); the problem with this approach in my opinion is that it seems unnecessarily complicated.

I went with the current approach because it seems both simple and resistant to future changes to the `Font` struct. I am open to other alternatives though of course.

Reply:
I think you can forget my last comment here, after https://github.com/jwilm/alacritty/pull/2836#discussion_r338853910 I do think that parameterizing `Reset` can resolve this (though not without a clone of the font still)